docs/specs/editorconfig-in-the-compiler.md

```diff
@@ -0,0 +1,111 @@
+All-capital words are interpreted per [RFC 2119](https://www.ietf.org/rfc/rfc2119.txt).
+
```

Comment:
We should probably add a section on compiler behavior for editorconfig.  In particular need to call out all of the forward thinking items we want controllable by editorconfig and what behavior the compiler needs now.

For example one item we want eventually is ability to control `#pragma` values.  That support needs to be added now or the compiler needs to completely reject any such annotations.  This is necessary in order to avoid compat breaks in the future when support is fully added.